tracing/src/main/java/com/palantir/tracing/TraceLocalObserver.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+import javax.annotation.Nullable;
+
+public interface TraceLocalObserver<T> {
```

Comment:
We could name this “Observer” and enclose it within TraceLocal. Not a strong opinion either way, but it may help with discovery.

Reply:
yeah fair enough - done